core/src/main/java/org/jboss/pnc/build/finder/core/BuildFinder.java

```diff
@@ -804,318 +608,6 @@ private String handleFileNotFound(String filename) {
-
-        allPncBuilds.putAll(allPncBuildsTemp);
-
-        it = artifactsList.iterator();
-
-        for (Entry<Checksum, Collection<String>> entry : entries) {
-            Checksum checksum = entry.getKey();
-            Collection<String> filenames = entry.getValue();
-            List<Artifact> artifacts;
-
-            if (pncChecksumCaches != null) {
-                artifacts = pncChecksumCaches.get(ChecksumType.md5).get(checksum.getValue());
-
-                if (artifacts != null) {
-                    LOGGER.debug("Found {} in Pnc checksum cache", checksum);
-                }
-            } else {
-                artifacts = it.next();
-            }
-
-            if (LOGGER.isDebugEnabled()) {
-                LOGGER.debug("PNC: number of artifacts={}", artifacts == null ? 0 : artifacts.size());
-            }
-
-            Artifact artifact = null;
-
-            if (artifacts != null && !artifacts.isEmpty()) {
-                artifact = getBestPncArtifact(artifacts);
-            }
-
-            if (artifact != null) {
-                List<Integer> buildIds = artifact.getBuildRecordIds();
-                Integer buildId;
-
-                if (!buildIds.isEmpty()) {
-                    buildId = buildIds.get(0);
-                } else {
-                    notFoundChecksums.put(checksum, filenames);
-                    continue;
-                }
-
-                LOGGER.debug("PNC: artifact={}, buildId={}", artifact, buildId);
-                PncBuild pncbuild = null;
-
-                if (pncBuildCache != null) {
-                    pncbuild = pncBuildCache.get(buildId);
-
-                    if (pncbuild != null) {
-                        LOGGER.debug("Found {} in Pnc build cache", buildId);
-                    }
-                }
-
-                if (pncbuild == null) {
-                    pncbuild = allPncBuilds.get(buildId);
-
-                    if (pncBuildCache != null && pncbuild != null) {
-                        pncBuildCache.put(pncbuild.getBuildRecord().getId(), pncbuild);
-                    }
-                }
-
-                if (pncbuild != null) {
-                    pncbuild.getArtifacts().add(artifact);
-                    BuildRecord record = pncbuild.getBuildRecord();
-                    Integer id = record.getId();
-                    KojiBuild kojibuild = builds.get(new BuildSystemInteger(id, BuildSystem.pnc));
-
-                    if (kojibuild == null) {
-                        kojibuild = PncUtils.pncBuildToKojiBuild(pncbuild);
-                        builds.put(new BuildSystemInteger(id, BuildSystem.pnc), kojibuild);
-                    }
-
-                    KojiArchiveInfo kojiarchive = PncUtils.artifactToKojiArchiveInfo(pncbuild, artifact);
-
-                    PncUtils.fixNullVersion(kojibuild, kojiarchive);
-
-                    addArchiveToBuild(kojibuild, kojiarchive, filenames);
-
```

Comment:
@dwalluck How are these 3 lines meant to work? The conversion doesn't work to me as the archives are not added to the build due to this code:
        Optional<KojiLocalArchive> matchingArchive = build.getArchives()
                .stream()
                .filter(a -> a.getArchive().getArchiveId().equals(archive.getArchiveId()))
                .findFirst();

The PncUtils.artifactToKojiArchiveInfo doesn't add the archives to the build and then it is filtered by this value in the addArchiveToBuild